File: VB/02_Text/FindAndHighlightText.vb
Imports Spire.Pdf
Imports Spire.Pdf.Texts

Namespace FindAndHighlightText
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new instance of PdfDocument
            Dim pdf As New PdfDocument()

            ' Load the PDF file from the specified path
            pdf.LoadFromFile("..\..\..\..\..\..\Data\FindAndHighlightText.pdf")

            ' Iterate through each page in the document
            For Each page As PdfPageBase In pdf.Pages

                ' Create a new instance of PdfTextFinder for the current page
                Dim finder As New PdfTextFinder(page)

                ' Set the search parameter to match whole word occurrences
                finder.Options.Parameter = TextFindParameter.WholeWord

                ' Find all occurrences of "science" in the current page
                Dim finds As List(Of PdfTextFragment) = finder.Find("science")

                ' Highlight each found occurrence
                For Each find As PdfTextFragment In finds
                    find.HighLight()
                Next find
            Next page

            ' Specify the output file path
            Dim output As String = "FindAndHighlightText_out.pdf"

            ' Save the file
            pdf.SaveToFile(output, FileFormat.PDF)

            ' Close the document
            pdf.Close()

            ' Launch the result file
            PDFDocumentViewer(output)
        End Sub

        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub
    End Class
End Namespace

File: CS/02_Text/FindAndHighlightText.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Texts;

namespace FindAndHighlightText
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Load the document from disk
            PdfDocument pdf = new PdfDocument();
            pdf.LoadFromFile(@"..\..\..\..\..\..\Data\FindAndHighlightText.pdf");

            // Iterate through each page of the document
            foreach (PdfPageBase page in pdf.Pages)
            {
                // Create a PdfTextFinder object for the current page
                PdfTextFinder finder = new PdfTextFinder(page);
                finder.Options.Parameter = Spire.Pdf.Texts.TextFindParameter.WholeWord;

                // Find the occurrences of the specified text
                List<PdfTextFragment> finds = finder.Find("science");

                // Highlight the found text
                foreach (PdfTextFragment find in finds)
                {
                    find.HighLight();
                }
            }

            // Save the modified document
            string output = "FindAndHighlightText_out.pdf";
            pdf.SaveToFile(output, FileFormat.PDF);

            //Launch the result file
            PDFDocumentViewer(output);
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
